File: nodejs/src/conDependencyInjection/conLibreria/UserController.js
export class UserController {

    constructor(opts) {
        this.userService = opts.userService;
    }

    create(ctx) {
        const {nombre, apellido} = ctx.req.query;
        const user = this.userService.create(nombre, apellido);
        ctx.res.json(user);
    }

    findById(ctx) {
        const {id} = ctx.req.query;
        const user = this.userService.findById(id)
        ctx.res.json(user);
    }
}

File: kotlin/src/main/kotlin/com/di/kotlin/di/spring/UserController.kt
package com.di.kotlin.di.spring

import com.di.kotlin.model.User
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Controller
import org.springframework.web.bind.annotation.PostMapping

@Controller("/di-solo")
class UserController {

    @Autowired private lateinit var  userService: UserService

    @PostMapping()
    fun getById(nombre: String, apellido: String): User {
        return this.userService.create(nombre, apellido)
    }
}